.github/workflows/pytest.yaml

```
@@ -26,24 +26,56 @@ jobs:
           # sudo rosdep init
           rosdep update
           rosdep install -i --from-paths . -y --rosdistro humble
-
       - name: Install dependent packages
         run: python3 -m pip install -r src/requirements.txt
 
       - name: Get package list
         run: |
           dpkg -l > package_list.txt
           python3 -m pip list > pip_list.txt
-
       - name: Upload package list
         uses: actions/upload-artifact@v3
         with:
           name: package_list
           path: |
             package_list.txt
             pip_list.txt
+      - name: Clone caret_analyze_cpp_impl
+        run: |
+          git clone https://github.com/tier4/caret_analyze_cpp_impl.git
+      - name: Restore Cache
+        id: cpp_impl-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/CARET_analyze_cpp_impl/**') }}
+
+      - name: Build caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        run: |
+          source /opt/ros/humble/setup.bash
+          cd caret_analyze_cpp_impl/
+          colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
+        shell: bash
+
+      - name: Cache caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/CARET_analyze_cpp_impl/**') }}
 
       - name: Run pytest
+        id: pytest_action
         run: |
           . /opt/ros/humble/setup.sh
+          source caret_analyze_cpp_impl/install/setup.bash
+          cd src
           python3 -m pytest
+        shell: bash
+
+      - name: If pytest failure
+        if: failure() && steps.pytest_action.outcome == 'failure'
+        run: |
+          echo "If pytest failure, are you sending both caret_analyze_cpp_impl and caret_analyze on this PR?"
+          echo "this pytest action may fail because it tests new caret_analyze in combination with old caret_analyze_cpp_impl."
```

Minor suggestion about message string.
```suggestion
          echo "Did you send PR to both caret_analyze and caret_analyze_cpp_impl?"
          echo "In that case, this pytest action may fail because it tests new caret_analyze in combination with old caret_analyze_cpp_impl."
```